hermes/bin/src/runtime_extensions/hermes/binary/mod.rs

```diff
@@ -1,14 +1,14 @@
 //! Binary runtime extension implementation.
 
-use crate::runtime_extensions::state::{Context, Stateful};
+use crate::runtime_extensions::state::Stateful;
 
 mod host;
 
 /// State
 pub(crate) struct State {}
 
 impl Stateful for State {
-    fn new(_ctx: &Context) -> Self {
+    fn new() -> Self {
         State {}
     }
 }
```

Comment:
I don't see the need for this trait because I don't think we have any generic/dynamic code that would require this. 

Each runtime extension `State` type can implement a function for creating an instance of it and we can just call it when creating the aggregate state (the one containing all extensions states) since it's all known at compile-time.

WDYT?

Reply:
yea, you are right at this moment it is not used as a trait (as a trait boundary and as a part of dynamic code).
But potentially we could need it later (or not, will see 😊).
Initially it was added as a additional reminder that for each runtime_extensions it is mandatory to add a `State` struct even if it is not contain anything.

Reply:
For now I would leave it as it is, it could be removed at any time later when we will have more mature state of Hermes.

Reply:
What might be cleaner however, is to just turn this into traditional functions.
As its probably always going to be necessary for any complex state to be fully managed inside the runtime module themselves, perhaps this is just cleaner if we have:

```rust
  fn new_context(ctx &context) {
      wasi::new_context(ctx);
      hermes::new_context(ctx);
  }

  // wasi module then has
  fn new_context(ctx &context) {
      cli::new_context(ctx);
      clocks::new_context(ctx);
      filesystem::new_context(ctx);
      http::new_context(ctx);
      io::new_context(ctx);
      random::new_context(ctx);
  }
   
 // hermes module has
 fn new_context(ctx &context) {
   binary::new_context(ctx);
   cardano::new_context(ctx);;
   // and so on...
 }

 // and so on....
```

This will make it explicit that we are not dealing with an actual memory allocation in the state, force each runtime extension to manage its own state given a particular context which it needs to do anyway and remove ambiguity.

This only gets called once when a new context is actually made (new app, module and execution count for the module changes).

Reply:
> Why did we remove the &Context? 
this Context (renamed to the `HermesRuntimeContext` inside `runtime_state.rs`) as a part of the `HermesRuntimeState`. `HermesRuntimeState` - actually initialised for every WASM execution and passed as a state to WASM.
So when we are talking about host implementation of the runtime extensions, they are using this `HermesRuntimeState` and could get a proper context how they are running.
But the runtime extension itself when we are defining it and initialising dont know and should not know about the this Context, because they are not running as a part of some WASM execution. They are separate.
```Rust
pub(crate) struct HermesRuntimeState {
    /// Runtime extensions state
    pub(crate) state: Arc<State>,
    /// Runtime context.
    pub(crate) _ctx: HermesRuntimeContext,
    /// Hermes event queue.
    pub(crate) _event_queue: Arc<HermesEventQueue>,
}
```

Reply:
@FelipeRosa from the latest discussion with @stevenj decided to remove this trait